### ISSUE TITLE
optimize spend lower/upper bounds held constant don't work when spend is not linear to avg media spend * n_time_periods

### ISSUE THREAD

**Issue by ohad-monday** [google/lightweight_mmm#161]
Im running the demo, and trying to keep one of the channels constant (with lower/upper bounds = 0).
It works fine when im using (btw i removed the extra features in the demo):
setting lower/upper bounds with 1st channel constant:
lower = jnp.repeat(0.2, mmm.n_media_channels)
upper = jnp.repeat(0.2, mmm.n_media_channels)
lower = lower.at[0].set(0)
upper = upper.at[0].set(0)
and (same as in the demo):
n_time_periods = 10
budget = jnp.sum(jnp.dot(prices, media_data.mean(axis=0)))* n_time_periods

works fine, channel 1 is same (previous=optimal)
<img width="865" alt="image" src="https://user-images.githubusercontent.com/61185809/215719444-ec3b1fa9-6f15-49ae-b295-b19840a7bbee.png">

BUT, when im using another budget () channel_1 is not constant (previous!=optimal)

<img width="854" alt="image" src="https://user-images.githubusercontent.com/61185809/215720048-921f21b1-2833-4c7f-a669-2022d641488b.png">

How can i keep channels constraints when the spend is not avg media spend * n_time_periods?

thanks!!

**Comment by michevan** [comment]
Thanks a lot for flagging this! I just wanted to share a link to our new "[support](https://github.com/google/lightweight_mmm#support)" section in the readme to give you some more information on how we can and cannot support such issues at the moment.

**Comment by ohad-monday** [comment]
@michevan any idea how to approach this? the readme doesn't help here

**Comment by ar-asur** [comment]
Hi @ohad-monday - Were you able to fix this? I am also facing similar issue where lower bound/upper bound per channel is not met in the optimization.
